webclient/src/components/Message/Message.tsx

```diff
@@ -0,0 +1,103 @@
+// eslint-disable-next-line
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { NavLink, generatePath } from "react-router-dom";
+
+import { RouteEnum } from 'types';
+
+import CardCallout from './CardCallout';
+import './Message.css';
+
+// eslint-disable-next-line
+const urlRegex = /(https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b(?:[-a-zA-Z0-9@:%_\+.~#?&//=]*))/g;
+const nameRegex = /(^[^: ]+):/;
+const mentionRegex = /(@[^ ]+)/g;
+const cardCalloutsRegex = /(\[\[[^\]]+\]\])/g;
+const calloutsRegex = /(\[\[|\]\])/g;
+
+const Message = ({ message: { message, messageType, timeOf, timeReceived } }) => (
+  <div className='message'>
+    <div className='message__detail'>
+      <ParsedMessage message={message} />
+    </div>
+  </div>
+);
+
+const ParsedMessage = ({ message }) => {
+  const [messageChunks, setMessageChunks] = useState(null);
+  const [name, setName] = useState(null);
+
+  useMemo(() => {
+    const name = message.match(nameRegex);
+
+    const messageChunks = message.replace(nameRegex, '')
+      .split(cardCalloutsRegex)
+      .filter(chunk => !!chunk)
+      .map(parseChunks);
+
+    if (name) {
+      setName(name[1]);
+    }
+
+    setMessageChunks(messageChunks);
+  }, [message]);
+
+  return (
+    <div>
+      { name && ( <strong><PlayerLink name={name} />:</strong> ) }
+      { messageChunks }
+    </div>
+  );
+};
+
+const PlayerLink = ({ name, label = name }) => (
+  <NavLink className="link" to={generatePath(RouteEnum.PLAYER, { name })}>
+    {label}
+  </NavLink>
+);
+
+function parseChunks(chunk, index) {
```

Comment:
This likewise is a bit of a hot mess, but I've done what i could to clean it up, and it's functional.

Reply:
Comments are all that's necessary to understand